ps.go

```diff
@@ -73,28 +82,45 @@ var psCommand = cli.Command{
 				continue
 			}
 			fields := strings.Fields(line)
-			p, err := strconv.Atoi(fields[pidIndex])
-			if err != nil {
-				return fmt.Errorf("unexpected pid '%s': %s", fields[pidIndex], err)
-			}
-
-			for _, pid := range pids {
-				if pid == p {
-					fmt.Println(line)
-					break
-				}
+			if fields[pidIndex] != cmdPs {
+				fmt.Println(line)
 			}
 		}
 		return nil
-	},
+	}
+
+	ns := fmt.Sprintf("/proc/%d/ns/pid", pids[0])
+	fd, err := syscall.Open(ns, syscall.O_RDONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("open /proc/%d/ns/pid %v", pids[0], err)
+	}
+	defer syscall.Close(fd)
+	err = system.Setns(uintptr(fd), 0)
```

Comment:
I still don't think we need to show inner pids for `runc ps`, the global pids would be more useful, and we can do `runc exec <id> ps -ef` if we really need to know the inner pids. Plus, `docker top` shows global pids all the time and it works well, no one complained about it, why we think runc users would like this?

Even if someone really like it, we should add option like `runc ps --inner <id>` instead of replacing the default.